docs/components/SideDrawer/Web.stories.tsx

```diff
@@ -268,6 +268,133 @@ const BackButtonTemplate: ComponentStory<typeof SideDrawer> = args => {
   );
 };
 
+const AnchoredTemplate: ComponentStory<typeof SideDrawer> = args => {
+  const [sideDrawerOpen, setSideDrawerOpen] = useState(args.open);
+  const anchorRef = useRef<HTMLDivElement>(null);
+
+  return (
+    <div style={{ display: "flex", gap: "24px" }}>
+      <div style={{ width: "200px", background: "#f5f5f5", padding: "16px" }}>
+        Left Panel
+      </div>
+      <div
+        ref={anchorRef}
+        style={{
+          width: "300px",
+          background: "#f5f5f5",
+          padding: "16px",
+          position: "relative",
+        }}
+      >
+        <Text>Middle Panel</Text>
+        <Button
+          onClick={() => setSideDrawerOpen(true)}
+          label="Open Side Drawer"
+        />
+      </div>
+      <div style={{ width: "200px", background: "#f5f5f5", padding: "16px" }}>
+        Right Panel
+      </div>
+
+      <SideDrawer
+        {...args}
+        open={sideDrawerOpen}
+        onRequestClose={() => setSideDrawerOpen(false)}
+        anchorElement={anchorRef}
+      >
+        <SideDrawer.Title>Anchored Side Drawer</SideDrawer.Title>
+        <Content>
+          <Text>
+            This drawer is anchored to the middle panel instead of the viewport.
+          </Text>
+        </Content>
+      </SideDrawer>
+    </div>
+  );
+};
+
+const CustomWidthAnchoredTemplate: ComponentStory<typeof SideDrawer> = args => {
+  const [sideDrawerOpen, setSideDrawerOpen] = useState(args.open);
+  const anchorRef = useRef<HTMLDivElement>(null);
+
+  // Calculate position manually to ensure proper alignment
+  const [position, setPosition] = useState({ left: 0 });
+
+  useEffect(() => {
```

Comment:
Could this logic be exposed in a hook, we seem to have something very similar in `useAnchorPosition`. 

Another thought could be allowing the width to be changed through a `width` prop